docs/src/api/params.md

```diff
@@ -398,6 +398,18 @@ set to `application/octet-stream` if not explicitly set.
 
 Whether to ignore HTTPS errors when sending network requests. Defaults to `false`.
 
+## js-python-fetch-option-follow
+* langs: js
+- `follow` <[int]>
+
+Maximum number of request allowed redirects. Defaults to `20`.
+
+## js-python-fetch-option-redirect
```

Comment:
I don't think we need 2 separate options for this and also 'error' mode can be turned into an assert on the call site. Let's introduce `maxRedirects` option instead, it would mean max number of redirect responses followed automatically. If set to 0 then the first response is always returned and that would cover the scenario requested in the bug.

Reply:
Hi, yes my first attempt was using only a `maxRedirects` option, but the default behavior is to fail when maxRedirects counter reach 0:

- When I set the option to a value > 0 then the right behavior is to throw an exception when `maxRedirects === 0`
- What if I want to obtain the exact initial 3xx request ('manual' case)? If I pass 0 as maxRedirect the same logic should apply and an exception is thrown without returning the desired response

Should I handle the zero case as 'don't follow the request' and still manage to let the function throw when I pass maxRedirects > 0 and the counter reach 0 or should I change the type to `'manual'  |  number` for example?

Reply:
Let's only throw in case `maxRedirects` option is not set. This will be equivalent to the current behavior and will throw in case of an infinite redirect. If `maxRedirects` is set the method will not throw, instead it will return last response in the chain even if it is another redirect. The client can alway assert that it has certain status code. This change won't break existing clients that don't set the option and will address the use case with redirect inspection.

Reply:
I added the `maxRedirects` option in the latest commit, let me know what you think, thanks